File: app/src/main/java/com/cashzhang/ashley/IndexItem.java
package com.cashzhang.ashley;

/**
 * Created by zhangchi on 2018/2/7.
 */

import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
import java.io.Serializable;

public
class IndexItem implements Serializable
{
    private static final long serialVersionUID = 200L;

    public long m_uid;
    public String m_url;
    public String[] m_tags;

    public
    IndexItem(long id, String url, String... tags)
    {
        m_uid = id;
        m_url = url;
        m_tags = tags;
    }

    private
    void writeObject(ObjectOutputStream out) throws IOException
    {
        out.writeLong(m_uid);
        out.writeUTF(m_url);
        out.writeObject(m_tags);
    }

    private
    void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException
    {
        m_uid = in.readLong();
        m_url = in.readUTF();
        m_tags = (String[]) in.readObject();
    }
}

File: app/src/main/java/com/cashzhang/nozdormu/IndexItem.kt
package com.cashzhang.nozdormu

import java.io.IOException
import java.io.ObjectInputStream
import java.io.ObjectOutputStream
import java.io.Serializable

/**
 * Created by cz21 on 2018/2/7.
 */
class IndexItem(var m_uid: Long, var m_url: String, vararg tags: String) : Serializable {
    var m_tags: Array<String>

    @Throws(IOException::class)
    private fun writeObject(out: ObjectOutputStream) {
        out.writeLong(m_uid)
        out.writeUTF(m_url)
        out.writeObject(m_tags)
    }

    @Throws(IOException::class, ClassNotFoundException::class)
    private fun readObject(`in`: ObjectInputStream) {
        m_uid = `in`.readLong()
        m_url = `in`.readUTF()
        m_tags = `in`.readObject() as Array<String>
    }

    companion object {
        private const val serialVersionUID = 200L
    }

    init {
        m_tags = tags
    }
}